### DISCUSSION TITLE
field for audio files upload and play

### DISCUSSION THREAD

**Discussion by theshubhamchugh** [filamentphp/filament#5999] [Ideas]
like have nova: https://nova.laravel.com/docs/4.0/resources/fields.html#audio-field

**Comment by VincvanAs** [comment]
Is this a question or just a way to say you prefer using nova? If you would like to upload and listen before submit, you have to code it by your own. For building an audio player in table e.g you could use for example htmlstring

**Comment by theshubhamchugh** [comment]
Thank you for your previous response. Based on your advice, I tried using FileUpload in FilamentPHP, and it seems to work well for my needs. However, as I am migrating my project from Nova to FilamentPHP and I am not an experienced developer, I encountered an issue with the file name being stored in the database in object form instead of a string format.

The output I am getting looks like "{"fada5b46-a235-4f95-af14-a665e2ff6eb3":"98c18b8c-6eee-4b50-b85b-d5255f30c518.mp3"}", but I want it to be stored as "mp3/98b583f9-44d3-40f8-8926-f24a4509c1f5.mp3" in the database.

Do you have any suggestions on how I can modify the output format to store the file name in a string format in the database? Thank you for your help.


code example: 

FileUpload::make('mp3')
                    ->disk('wasabi')
                    ->directory('mp3')
                    ->acceptedFileTypes(['audio/mpeg'])
                    ->maxSize(2000)
                    ->enableDownload()
                    ->visibility('public')
                    ->saveUploadedFileUsing(function () use ($uuid) {
                        return (string) str($uuid . '.mp3');
                    }),

**Comment by VincvanAs** [comment]
protected function mutateFormDataBeforeCreate(array $data): array
{
    $data['new_mp3'] = 'mp3/' . $data['mp3'];
 
    return $data;
}

https://filamentphp.com/docs/2.x/admin/resources/creating-records#customizing-data-before-saving
